Update dnsNameservers context to pick up values

dalees opened the pull request:
The `dnsNameservers` value has moved into `managedSubnets` section, but was trying to reference `.internalNetwork.dnsNameservers` which does not exist in chart values.

This change removes the context switch into `internalNetwork` to allow all references to correctly find their related values.

Closes: https://github.com/azimuth-cloud/capi-helm-charts/issues/476

dalees commented:
@mkjpryor If there is a better approach here than removing the context change `with`, please let me know. I'll be happy to update this PR.

It would be nice to move the `.Values` field, but I'm not sure how to do that in Helm, in a backwards compatible way.

mkjpryor commented:
@dalees 

I was also thinking that the fix here should probably be to move the values field into `internalNetwork`, since it is only respected by the managed subnet.

In order to maintain backwards compatibility, I think we just have to replace

```
{{- with .dnsNameservers }}
```

with

```
{{- with (default $.Values.clusterNetworking.dnsNameservers .dnsNameservers) }}
```

?

dalees commented:
Ah thanks, @mkjpryor - that was a lot easier than what I was thinking of.

I wasn't sure of the best practice around referencing values outside the current scope.

Updated the PR to use your suggestion and moved the values field.

mkjpryor commented on charts/openstack-cluster/values.yaml:
```diff
@@ -73,7 +73,8 @@ kubeNetwork:
 clusterNetworking:
   # Custom nameservers to use for the hosts
-  dnsNameservers:
+  # Deprecated; use `clusterNetworking.internalNetwork.dnsNameservers` instead.
+  dnsNameservers: []
```
I think just remove this completely

dalees replied:
Perhaps? It would keep the values file cleaner, but I note that the `magnum-capi-helm` driver still uses this - I figured it would be nicer to keep it documented? (Commenting out the actual field may help?)

This appears to be the practice [in other helm charts](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L4232), but happy to change if the code style here differs for a reason. I'm just interested in understanding more.

mkjpryor replied:
Maybe commented out is good.

dalees replied:
Done

mkjpryor commented on charts/openstack-cluster/values.yaml:
```diff
@@ -97,6 +98,10 @@ clusterNetworking:
     nodeCidr: 192.168.3.0/24
+    # List of nameserver IPs to use when creating a cluster network.
+    # This is only used if neither of networkFilter and subnetFilter are given
+    dnsNameservers: []
```
Can we default this to `null` as before, instead of `[]`?

dalees replied:
Annoying style question:

Functionally this is no different (`default` deals with empty lists as if they are `null`) but I felt it provided a hint to the user of the chart as to the type of field this is expected to be.

As before, happy to make the change just interested in why :) It would match the rest of the file better, so that's reason enough perhaps.

mkjpryor replied:
I was just going for consistency with the previous default, TBH. Not specifying it means "use the network default DNS servers", so I thought null was a better default than `[]` at the time. I'm not that bothered though.

sd109 reviewed (approved):
LGTM, thanks for the fix.
